### DISCUSSION TITLE
Failure (cannot get record ID: bad HTTP status: 403: {"success":false,"errors":[{"code":10000,"message":"Authentication error"}]})

### DISCUSSION THREAD

**Discussion by csdbz** [qdm12/ddns-updater#268] [Q&A]
Hi Quentin, thanks for making this. I'm not very competent with DNS (or Docker or Go or coding, generally), but I'm giving this a try. I got the container running and accessible, but not configured correctly--the webUI gives `Failure (cannot get record ID: bad HTTP status: 403: {"success":false,"errors":[{"code":10000,"message":"Authentication error"}]})`

Some uncertainties about my deployment lead to questions/clues: 

I didn't modify your docker-compose, but I added build arguments, `docker build -t qmcgaw/ddns-updater https://github.com/qdm12/ddns-updater.git --build-arg UID=501 --build-arg GID=502`. Were those args applied when I then ran `docker-compose -f "ddns-updater/docker-compose.yml" up -d --build`?

I'm on Cloudflare and I'm pretty sure I found the right Zone ID for `config.json`, but not sure I got the token right. Cloudflare's Token Summary says `All zones - Zone:Edit`, did I pick the right settings while creating the token?

I'm using a reverse proxy (Traefik), should I be using that docker network or should bridge work?

**Comment by qdm12** [comment]
Hey there!

> Were those args applied when I then ran

Yes, since you use the image `qmcgaw/ddns-updater` which you have built with the previous `docker build` command.
There is no point passing ` --build` with the existing docker-compose.yml. You could however define a build context and Dockerfile in the docker-compose.yml if you want, and in that case `--build` will re-build the image (when needed).

>  but not sure I got the token right

Yeah it might be wrong, since it responds with `403` (Forbidden). You could try running the program with `LOG_LEVEL=debug` and share the logs (remove your credentials), maybe I can help. Just in case you did not find it, there is a cloudflare readme [here](
https://github.com/qdm12/ddns-updater/blob/master/docs/cloudflare.md)

> should I be using that docker network or should bridge work?

It doesn't matter, up to you! As long as it has internet access
